app/src/pages/playground/PlaygroundChatTemplate.tsx

```diff
@@ -427,20 +458,23 @@ function SortableMessageItem({
             includeLabel={false}
             role={message.role}
             onChange={(role) => {
+              const content = message.content;
               let toolCalls = message.toolCalls;
               // Tool calls should only be attached to ai messages
               // Clear tools from the message and reset the message mode when switching away form ai
               if (role !== "ai") {
                 toolCalls = undefined;
-                setMessageMode("text");
+                setAIMessageMode("text");
               }
               updateInstance({
                 instanceId: playgroundInstanceId,
                 patch: {
                   template: {
                     __type: "chat",
                     messages: template.messages.map((msg) =>
-                      msg.id === message.id ? { ...msg, role, toolCalls } : msg
+                      msg.id === message.id
+                        ? { ...msg, role, toolCalls, content }
+                        : msg
```

Comment:
why does content need to be added here as well now? instead of just spread on message